Flagsmith.FlagsmithClient/FlagsmithClient.cs

```diff
@@ -28,217 +28,190 @@ namespace Flagsmith
             }
         }
 
+        ~FlagsmithClient()
+        {
+            _pollingManager?.StopPoll();
+        }
+
         /// <summary>
         /// Get all the default for flags for the current environment.
         /// </summary>
-        public async Task<Flags> GetEnvironmentFlags()
-            => Environment != null ? GetFeatureFlagsFromDocuments() : await GetFeatureFlagsFromApi();
+        public async Task<IFlags> GetEnvironmentFlags()
+            => _environment != null ? GetFeatureFlagsFromDocuments() : await GetFeatureFlagsFromApi();
 
         /// <summary>
         /// Get all the flags for the current environment for a given identity.
         /// </summary>
-        public async Task<Flags> GetIdentityFlags(string identity)
+        public async Task<IFlags> GetIdentityFlags(string identity)
         {
             return await GetIdentityFlags(identity, null);
         }
 
         /// <summary>
         /// Get all the flags for the current environment for a given identity with provided traits.
         /// </summary>
-        public async Task<Flags> GetIdentityFlags(string identity, List<Trait> traits)
+        public async Task<IFlags> GetIdentityFlags(string identity, IEnumerable<ITrait> traits)
         {
-            if (Environment != null)
+            if (_environment != null)
             {
                 return GetIdentityFlagsFromDocuments(identity, traits);
             }
-
             return await GetIdentityFlagsFromApi(identity, traits);
         }
 
-        public List<Segment> GetIdentitySegments(string identifier)
+        public Task<IReadOnlyCollection<ISegment>> GetIdentitySegments(string identifier)
         {
-            return GetIdentitySegments(identifier, new List<Trait>());
+            return GetIdentitySegments(identifier, Array.Empty<ITrait>());
         }
 
-        public List<Segment> GetIdentitySegments(string identifier, List<Trait> traits)
+        public Task<IReadOnlyCollection<ISegment>> GetIdentitySegments(string identifier, IEnumerable<ITrait> traits)
         {
-            if (this.Environment == null)
+            if (_environment == null)
             {
                 throw new FlagsmithClientError("Local evaluation required to obtain identity segments.");
             }
-            IdentityModel identityModel = new IdentityModel { Identifier = identifier, IdentityTraits = traits?.Select(t => new TraitModel { TraitKey = t.GetTraitKey(), TraitValue = t.GetTraitValue() }).ToList() };
-            List<SegmentModel> segmentModels = Evaluator.GetIdentitySegments(
-                this.Environment, identityModel, new List<TraitModel>()
-            );
-
-            return segmentModels?.Select(t => new Segment(id: t.Id, name: t.Name)).ToList();
+            var identityModel = new IdentityModel { Identifier = identifier, IdentityTraits = traits?.Select(t => new TraitModel { TraitKey = t.Key, TraitValue = t.Value }).ToList() };
```

Comment:
What's the benefit to being less explicit with the variable typing here?

Reply:
less typing :) the type is already in "new ....". using "var" is generally accepted and has also advantages when making refactors